### ISSUE TITLE
Ignore Terragrunt errors in Infracost CLI output

### ISSUE THREAD

**Issue by sfenman** [infracost/infracost#994]
Having a big directory with many Terragrunt modules and use run-all in this directory can lead to errors if some of the modules are not configured properly. For example I have this project with 4 different modules, one of them ( errordir ) with badly configured code:

```
.
├── ecr
│   └── terragrunt.hcl
├── ecs-service
│   └── terragrunt.hcl
├── ecs-taskdef
│   └── terragrunt.hcl
└── errordir
    └── terragrunt.hcl
```
Running `infracost diff --path=.`

Output:

```
❯ infracost diff --path=.
Detected Terragrunt directory at .
  ✖ Running terragrunt run-all terragrunt-info 

  Terragrunt command failed with:
    time=2021-09-30T17:41:21+03:00 level=error msg=Error processing module at '/home/sfenman/ecs-services/api/errordir/terragrunt.hcl'. How this module was found: Terragrunt config file found in a subdirectory of /home/sfenman/ecs-services/api. Underlying error: /home/sfenman/ecs-services/api/errordir/terragrunt.hcl:1,1-6: Argument or block definition required; An argument or block definition is required here. To set an argument, use the equals sign "=" to introduce the argument value.
    time=2021-09-30T17:41:21+03:00 level=error msg=Unable to determine underlying exit code, so Terragrunt will exit with error code 1

Error: exit status 1
```

Infracost can print but ignore these errors and then show output for the other modules.

**Comment by aliscott** [comment]
@sfenman thanks for the issue. We use the `run-all` for discovering the directories and also for running `terragrunt plan`. I can't find a way to ignore errors when running `terragrunt run-all`. Do you know if there's a way?

Another option I can think of is allowing the [`--terragrunt-exclude-dir`](https://terragrunt.gruntwork.io/docs/reference/cli-options/#terragrunt-exclude-dir) flag to be passed through. Unfortunately terragrunt don't provide an env var for this, so we'd need to find some way of passing the flag through. Would that work for this use case?

**Comment by sfenman** [comment]
Yes that would be really good. An even better solution would be a `.ignore` file along with the option to pass an `exlude-dir` flag.

**Comment by alikhajeh1** [comment]
@sfenman did you end-up finding a work-around for this? It's tricky for us to action as @aliscott mentioned.
Do you think this is a common issue for Terragrunt users? One way we could get around this might be to parse the HCL code (instead of using plan JSONs) but that's complicated.

**Comment by sfenman** [comment]
Unfortunately I haven't found a workaround, but then again I don't think it's a major problem since most of the time I'm applying Infracost to smaller modules and I guess the majority of Terragrunt users does too.

**Comment by alikhajeh1** [comment]
@sfenman good news! We're releasing https://github.com/infracost/docs/pull/133/files in v0.9.15 next week so you can exclude dirs that have issues. Does that help? If so, can we close this issue?

**Comment by sfenman** [comment]
It does help, thanks a lot! Feel free to close the issue.

**Comment by alikhajeh1** [comment]
Awesome! (thanks @tim775!)

**Comment by alikhajeh1** [comment]
@sfenman the `INFRACOST_TERRAGRUNT_FLAGS` env variable was released a few hours ago in Infracost version v0.9.15.

You can check out the docs for it here https://www.infracost.io/docs/iac_tools/terragrunt/#terragrunt-cli-options. This now gives a way of passing the `--terragrunt-exclude-dir` option through to Infracost.
